partner_state/models/partner.py

```diff
@@ -141,9 +145,8 @@ def message_track(self, tracked_fields, initial_values):
         from field properties to make message
         """
         # TODO we should use company of modified partner
-        for line in self.env['res.partner.state_field'].search([]):
-            if line.track:
-                field = self._fields[line.field_id.name]
-                setattr(field, 'track_visibility', 'always')
+        for line in self.env['res.partner.state_field'].filtered('line.track'):
```

Comment:
The filtered is badly used in this case, because you do not have any recordset to filter. Use it like this self.env['res.partner.state_field'].search([]).filtered('track')